### DISCUSSION TITLE
Hello   I need help to increase max duration

### DISCUSSION THREAD

**Discussion by pena1973** [vercel/community#6684] [Help] [answered]
Hello,
I'm reaching out to the support team as I need to set up maxDuration, but I'm having trouble doing so. I've read the documentation at https://vercel.com/docs/functions/configuring-functions/duration and tried to apply it, but I'm encountering issues.
I have a backend that receives requests from the frontend and processes them, while also making requests to external services. The routing is handled through Express, and the project is in TypeScript.
Here's how my server initialization looks:
```
export default function initServer(): Express {
  const app = express();
///…….  

  const jsonMiddleware = express.json();
  app.use(jsonMiddleware);
  
//  Here are my routes which occasionally result in timeouts (status code 504)
  app.use("/play", tokenVerify, playRouter);
  app.use("/text", tokenVerify, textRouter);
  app.use("/user", tokenVerify, userRouter);
  app.use("/adv",  advRouter);  
  app.use("/auth", authRouter);
  app.use("/regl", reglRouter);

  app.use("/", (_, res) => {
    res.send("Server lounched!");
  });

  app.listen(port, () => {
    console.log(`Server running on port ${port}`);
  });

  return app;
}
```

And here's how my function file looks:
```
import { Request, Response, Router } from 'express';
export const advRouter = Router();

// here I try to set maxDuration – zero effect
export const maxDuration = 60;
export const config = { maxDuration: 299 }

advRouter.get('/', async (req: Request, res: Response) => {
  try {
///  my code
    res.send("adv");
  } catch (e) { 
    res.status(500);
    res.send("Something went wrong");
  }
});
```

I'm trying to configure the vercel.json file
This doesn't work because builds and functions cannot be used at the same time
```
{
  "version": 2,
  "builds": [
    {
      "src": "public/index.js",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "/public/index.js"
    }
  ],
  
    "functions": {
      "src/api/**/*": {
        "maxDuration": 60
      }
    }
  
}
```

// so when I deploy with file below  I just get a file with code as deploy result
```
{
  "version": 2,
  "routes": [
    {
      "src": "/(.*)",
      "dest": "/public/index.js"
    }
  ],
  
    "functions": {
      "src/api/**/*": {
        "maxDuration": 60
      }
    }  
}
```

When deploying, Vercel doesn't detect the serverless functions, probably because I'm using a router.

Can you provide an example of how to set maxDuration in this situation? I upgraded to the PRO plan just for this functionality, so any assistance would be greatly appreciated.
.

**Comment by amyegan** [comment]
Hey @pena1973. I would recommend updating the legacy [`builds`](https://vercel.com/docs/projects/project-configuration#builds)  property to use the [`functions`](https://vercel.com/docs/projects/project-configuration#functions) property instead. That will resolve the error message about combining `builds` and `functions`. You should also consider updating the legacy [`routes`](https://vercel.com/docs/projects/project-configuration#routes) property.

> The routes property is only meant to be used for advanced integration purposes, such as the [Build Output API](https://vercel.com/docs/build-output-api/v3), and cannot be used in conjunction with any of the properties mentioned above.

It looks like you're using something other than Next.js for this project, so adding `export const maxDuration = 60;` won't have any effect. But you'll be able to set a higher duration in `vercel.json` if you upgrade the legacy configuration properties.

**Comment by Thribs** [comment]
Having the same issue. Moving from express to a functions declaration structure would be a huge task, since I depend heavily on Express middleware functions. Any updates on this?

**Comment by Thribs** [comment]
With help from [this video](https://www.youtube.com/watch?v=B-T69_VP2Ls) I was able to do it.

tl;dr we need:
 - a file that imports then exports the express app
 - a rewrite on the vercel.json configuration to that file
 - overriding vercel-build command so no build happens(that would hang the server and never finish the build step)
 
 final vercel.json file:
 ```
{
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/api/server.js"
    }
  ],
  "functions": {
    "api/server.js": {
      "maxDuration": 30
    }
  }
}
```

**Reply by pena1973** [reply, marked as answer]
I went to digital ocean and all works  and it is cheeper
